crates/core/src/transcript/mod.rs

```diff
@@ -434,7 +434,7 @@ where
 fn sample_bits_reader<Reader: Buf>(mut reader: Reader, bits: usize) -> usize {
```

Comment:
It'd be better to now just return `-> u32` and adjust the line below accordingly.

Reply:
this leads to a big propagation of type changes; are you ok with that?

Reply:
can you take a look at dbe9836? i opted to convert it back to `usize` after the return. we could further redo things so that all of `log_dim`, `fold_arities`, etc., were all `u32`s, instead of `usize`s; but this leads to a pretty big propagation of type changes? let me know your thoughts

Reply:
LGTM now